src/test/java/com/google/cloud/flink/bigquery/integration/FlinkBQWriteIntegrationTest.java

```diff
@@ -0,0 +1,307 @@
+            + "',\n"
+            + "  'filter' = 'word_count > 500',\n"
+            + "  'credentialsFile' = '"
+            + System.getenv("GOOGLE_APPLICATION_CREDENTIALS")
+            + "' ,\n"
+            + "  'selectedFields' = 'word,word_count' \n"
+            + ")");
+    Table sourceTable = tEnv.from(flinkSrcTable);
+    Table sourceResultTable =
+        sourceTable.where($("word_count").isGreaterOrEqual(500)).select($("word"), $("word_count"));
+    BigQueryTableSink tableSink = new BigQueryTableSink(sourceResultTable, bigqueryWriteTable);
+
+    tEnv.registerTableSink(flinkSinkTable, tableSink);
+
+    TableResult sinkResult = sourceResultTable.executeInsert(flinkSinkTable, true);
+    JobExecutionResult jobExecutionResult =
+        sinkResult
+            .getJobClient()
+            .get()
+            .getJobExecutionResult(Thread.currentThread().getContextClassLoader())
+            .get();
+  }
+
+  @Test
+  public void testPartitionHourly() throws Exception {
+    String bigqueryReadTable = "bigquery-public-data.stackoverflow.posts_questions";
+    String bigqueryWriteTable =
+        PROJECT_ID + "." + testDataset.toString() + "." + "hourlyPartitioned";
+    String filter = "view_count=300 and owner_user_id<5000";
+    testPartition("HOUR", bigqueryReadTable, bigqueryWriteTable, filter);
+    Thread.sleep(120 * 1000);
+    int count = bqReadTableForPartiton(bigqueryWriteTable);
+    assertThat(count).isEqualTo(37);
+  }
+
+  @Test
+  public void testPartitionDaily() throws Exception {
+    String bigqueryReadTable = "bigquery-public-data.stackoverflow.posts_questions";
+    String bigqueryWriteTable =
+        PROJECT_ID + "." + testDataset.toString() + "." + "dailyPartitioned";
+    String filter = "view_count=300 and owner_user_id<5000";
+    testPartition("DAY", bigqueryReadTable, bigqueryWriteTable, filter);
+    Thread.sleep(120 * 1000);
+    int count = bqReadTableForPartiton(bigqueryWriteTable);
+    assertThat(count).isEqualTo(37);
+  }
+
+  @Test
+  public void testPartitionMonthly() throws Exception {
+    String bigqueryReadTable = "bigquery-public-data.stackoverflow.posts_questions";
+    String bigqueryWriteTable =
+        PROJECT_ID + "." + testDataset.toString() + "." + "monthlyPartitioned";
+    String filter = "view_count=300 and owner_user_id<5000";
+    testPartition("MONTH", bigqueryReadTable, bigqueryWriteTable, filter);
+    Thread.sleep(120 * 1000);
+    int count = bqReadTableForPartiton(bigqueryWriteTable);
+    assertThat(count).isEqualTo(37);
+  }
+
+  @Test
+  public void testPartitionYearly() throws Exception {
+    String bigqueryReadTable = "bigquery-public-data.stackoverflow.posts_questions";
+    String bigqueryWriteTable =
+        PROJECT_ID + "." + testDataset.toString() + "." + "yearlyPartitioned";
+    String filter = "view_count=300 and owner_user_id<5000";
+    testPartition("YEAR", bigqueryReadTable, bigqueryWriteTable, filter);
+    Thread.sleep(120 * 1000);
```

Comment:
Why the sleep is needed?

Reply:
Takes some amount of time to write the data. Same wait time found in spark connector also